test_rclcpp/test/test_multiple_service_calls.cpp

```diff
@@ -69,4 +71,60 @@ TEST(CLASSNAME(test_multiple_service_calls, RMW_IMPLEMENTATION), multiple_servic
   rclcpp::spin_until_future_complete(node, result2);
   printf("Received second reply\n");
   EXPECT_EQ(2, result2.get()->sum);
+  // The first result should still be 1.
+  EXPECT_EQ(1, result1.get()->sum);
+}
+
+TEST(CLASSNAME(test_multiple_service_calls, RMW_IMPLEMENTATION), multiple_clients) {
+  uint32_t n = 5;
+
+  rclcpp::init(0, nullptr);
+
+  auto node = rclcpp::Node::make_shared("test_multiple_clients");
+
+  node->create_service<test_rclcpp::srv::AddTwoInts>(
+    "test_multiple_clients", handle_add_two_ints);
+
+  using ClientRequestPair = std::pair<
+      rclcpp::client::Client<test_rclcpp::srv::AddTwoInts>::SharedPtr,
+      test_rclcpp::srv::AddTwoInts::Request::SharedPtr>;
+  using SharedFuture = rclcpp::client::Client<test_rclcpp::srv::AddTwoInts>::SharedFuture;
+
+  std::vector<ClientRequestPair> client_request_pairs;
+  for (uint32_t i = 0; i < n; ++i) {
+    auto client = node->create_client<test_rclcpp::srv::AddTwoInts>("test_multiple_clients");
+    auto request = std::make_shared<test_rclcpp::srv::AddTwoInts::Request>();
+    request->a = i;
+    request->b = i + 1;
+    client_request_pairs.push_back(ClientRequestPair(client, request));
+  }
+
+  std::vector<SharedFuture> results;
+  // Send all the requests
+  for (auto & pair : client_request_pairs) {
+    results.push_back(pair.first->async_send_request(pair.second));
+  }
+
+  printf("Sending %u requests...\n", n);
+  fflush(stdout);
+
+  auto timer_callback = [&results]() {
+      bool all_ready = true;
+      for (auto & result : results) {
+        all_ready &= result.wait_for(std::chrono::seconds(0)) == std::future_status::ready;
+      }
+      if (all_ready) {
+        rclcpp::shutdown();
+      }
+    };
+  auto timer = node->create_wall_timer(std::chrono::nanoseconds(3000000), timer_callback);
+
+  rclcpp::spin(node);
+
+  // Check the status of all futures
+  for (uint32_t i = 0; i < results.size(); i++) {
+    printf("Got request #%u\n", i);
```

Comment:
The result has not been received yet. This print should be moved one line down.

Also flush afterwards?